File: backend/internal/token/token.go
package token

type TokenType string

const (
	PLUS     = "+"
	MINUS    = "-"
	ASTERISK = "*"
	SLASH    = "/"
	CARET    = "^"

	EQUALS     = "="
	NOT_EQUALS = "!="
	LT         = "<"
	LE         = "<="
	GT         = ">"
	GE         = ">="

	BANG = "!"

	COMMA  = ","
	LPAREN = "("
	RPAREN = ")"

	IDENT  = "IDENT"
	NUMBER = "NUMBER"

	NEW_LINE = "NEW_LINE"
	EOF      = "EOF"
	ILLEGAL  = "ILLEGAL"
)

type Token struct {
	Type    TokenType
	Literal string
}

func New(tokenType TokenType, literal string) Token {
	return Token{
		Type:    tokenType,
		Literal: literal,
	}
}

func (tok *Token) IsSymbol() bool {
	return tok.Type == IDENT && len(tok.Literal) == 1
}

File: backend/internal/tokens/token.py
from enum import Enum
from typing import NamedTuple


class TokenType(Enum):
    PLUS = "+"
    MINUS = "-"
    ASTERISK = "*"
    SLASH = "/"
    CARET = "^"

    EQUALS = "="
    NOT_EQUALS = "!="
    LT = "<"
    LE = "<="
    GT = ">"
    GE = ">="

    BANG = "!"

    COMMA = ","
    LPAREN = "("
    RPAREN = ")"

    IDENT = "IDENT"
    NUMBER = "NUMBER"

    NEW_LINE = "NEW_LINE"
    EOF = "EOF"
    ILLEGAL = "ILLEGAL"


class Token(NamedTuple):
    ttype: TokenType
    literal: str

    def __repr__(self):
        return f"Token({self.ttype}, {self.literal})"

    def __eq__(self, value) -> bool:
        return (
            isinstance(value, Token)
            and self.ttype == value.ttype
            and self.literal == value.literal
        )

    def is_symbol(self) -> bool:
        return self.ttype == TokenType.IDENT and len(self.literal) == 1
